src/main/java/DialogBox.java

```diff
@@ -0,0 +1,85 @@
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+import javafx.scene.shape.Circle;
+
+/**
+ * Represents a window controller for dialog box using FXML.
+ * Consists of an ImageView to represent the speaker's face and
+ * a label containing text from the speaker.
+ */
+public class DialogBox extends HBox {
+    @FXML
+    private Label dialog;
+
+    @FXML
+    private ImageView displayPicture;
+
+    /**
+     * Constructs DialogBox object.
+     *
+     * @param text speaker's message.
+     * @param img speaker's image.
+     */
+    private DialogBox(String text, Image img) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBox.fxml"));
+            fxmlLoader.setController(this);
+            fxmlLoader.setRoot(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        dialog.setText(text);
+        displayPicture.setImage(img);
+
+        Circle clip = new Circle(50, 50, 48);
+        displayPicture.setClip(clip);
+    }
+
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
+        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
+        Collections.reverse(tmp);
+        getChildren().setAll(tmp);
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    /**
+     * Gets DialogBox object for User speaker.
+     *
+     * @param text speaker's message.
+     * @param img speaker's image.
+     * @return new dialog object.
+     */
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
+    /**
+     * Gets DialogBox object for Duke speaker.
+     * Flips speaker's image horizontally.
+     *
+     * @param text speaker's message.
+     * @param img speaker's image.
+     * @return new dialog object.
+     */
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```

Comment:
Do you want to specify the datatype of db in its declaration instead of using `var`?